apps/console/src/features/roles/components/edit-role/edit-role-federated-groups-accordion.tsx

```diff
@@ -0,0 +1,449 @@
+        useCallback(debounce((query: string) => {
+            if (isEmpty(query)) {
+                setGroupsOptions(originalGroupList);
+                setGroupSearchLoading(false);
+
+                return;
+            }
+            
+            const trimmedQuery: string = query.trim();
+            const filteredGroups: IdentityProviderGroupInterface[] = originalGroupList?.filter(
+                (group: IdentityProviderGroupInterface) => {
+                    return group.name.toLowerCase().includes(trimmedQuery.toLowerCase());
+                });
+                
+            setGroupsOptions(filteredGroups);
+            setGroupSearchLoading(false);
+        }, RoleConstants.DEBOUNCE_TIMEOUT), [ originalGroupList ]);
+
+    /**
+     * Handle the restore groups.
+     * 
+     * @param remainingGroups - remaining groups.
+     */
+    const handleRestoreGroups = (remainingGroups: IdentityProviderGroupInterface[]) => {
+        const removedGroups: IdentityProviderGroupInterface[] = [];
+
+        removedGroupsOptions.forEach((group: IdentityProviderGroupInterface) => {
+            if (!remainingGroups?.find((newGroup: IdentityProviderGroupInterface) => newGroup.id === group.id)) {
+                removedGroups.push(group);
+            }
+        });
+
+        setSelectedGroupsOptions([
+            ...selectedGroupsOptions,
+            ...removedGroups
+        ]);
+    };
+
+    /**
+     * Resolves the IDP image.
+     * 
+     * @returns Resolved IDP image.
+     */
+    const resolveIDPImage = (): string => {
+        if (identityProvider.image) {
+            return ConnectionsManagementUtils.resolveConnectionResourcePath(
+                "", identityProvider.image);
+        }
+        
+        return getConnectionIcons().default;
+    };
+
+    return (
+        <EmphasizedSegment
+            className="mt-2"
+            basic
+        >
+            <Accordion
+                onChange={ onAccordionChange }
+                elevation={ 1 }
+                variant="elevation"
+                disableGutters
+                expanded={ isExpanded }
+            >
+                <AccordionSummary
+                    aria-controls="panel1a-content"
+                    id="panel1a-header"
+                    expandIcon={ <ChevronDownIcon /> }
+                >
+                    <>
+                        <GenericIcon
+                            square
+                            transparent
+                            className="card-image"
+                            size="mini"
+                            icon={ resolveIDPImage() }
+                            floated="left"
+                        />
+                        <Typography
+                            className="ml-4"
+                            variant="h6"
+                        >
+                            { identityProvider.name }
+                        </Typography>
+                    </>
+                    
```

Comment:
Let's remove the redundant line.

Reply:
Addressed 5ce5a869d09cb34de9eac44ddee5d1053e6006a4